sycl/test-e2e/ESIMD/unified_memory_api/Inputs/prefetch.hpp

```diff
@@ -279,3 +279,246 @@ template <typename T> bool testUSM(queue Q) {
+               if constexpr (sizeof(T) >= 4) {
+                 if (GlobalID % 4 == 0) // ByteOffset - simd
+                   prefetch<T>(InAcc, ByteOffsets, Props);
+                 else if (GlobalID % 4 == 1)
+                   prefetch<T>(InAcc, Props);
+                 else if (GlobalID % 4 == 2)
+                   prefetch<T>(InAcc, ByteOffset, Props);
+                 else // ByteOffset - simd_view
+                   prefetch<T, N>(InAcc, ByteOffsetsView, Props);
+               } else {
+                 if (GlobalID % 2 == 0) // ByteOffset - simd
+                   prefetch<T>(InAcc, ByteOffsets, Props);
+                 else // ByteOffset - simd_view
+                   prefetch<T, N>(InAcc, ByteOffsetsView, Props);
+               }
+             } else {
+               if constexpr (sizeof(T) >= 4) {
+                 if (GlobalID % 4 == 0) // ByteOffset - simd
+                   prefetch<T>(InAcc, ByteOffsets);
+                 else if (GlobalID % 4 == 1)
+                   prefetch<T>(InAcc);
+                 else if (GlobalID % 4 == 2)
+                   prefetch<T>(InAcc, ByteOffset);
+                 else // ByteOffset - simd_view
+                   prefetch<T, N>(InAcc, ByteOffsetsView);
+               } else {
+                 if (GlobalID % 2 == 0) // ByteOffset - simd
+                   prefetch(In, ByteOffsets);
+                 else // ByteOffset - simd_view
+                   prefetch<T, N>(InAcc, ByteOffsetsView);
+               }
+             }
+           }
+         } // end if (VS == 1)
+         Vals = gather<T, N, VS>(InAcc, ByteOffsets);
+         Vals.copy_to(Out + GlobalID * N);
+       });
+     }).wait();
+  } catch (sycl::exception const &e) {
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+    sycl::free(In, Q);
+    sycl::free(Out, Q);
+    return false;
+  }
+
+  bool Passed = verify(In, Out, N, Size, VS);
+  if (!Passed)
+    std::cout << "Case FAILED" << std::endl;
+
+  sycl::free(In, Q);
+  sycl::free(Out, Q);
+  return Passed;
+}
+
+template <typename T> bool testACC(queue Q) {
+  constexpr bool UseMask = true;
+  constexpr bool UseProperties = true;
+
+  properties CacheProps{cache_hint_L1<cache_hint::streaming>,
+                        cache_hint_L2<cache_hint::cached>};
+
+  bool Passed = true;
+  Passed &= testACC<T, 1, 1, !UseMask, !UseProperties>(Q, 2, CacheProps);
+  Passed &= testACC<T, 2, 1, UseMask, !UseProperties>(Q, 2, CacheProps);
+  Passed &= testACC<T, 4, 1, UseMask, !UseProperties>(Q, 2, CacheProps);
+  Passed &= testACC<T, 8, 1, UseMask, UseProperties>(Q, 3, CacheProps);
+
+  Passed &= testACC<T, 1, 1, !UseMask, UseProperties>(Q, 2, CacheProps);
+  Passed &= testACC<T, 2, 1, UseMask, UseProperties>(Q, 2, CacheProps);
+  Passed &= testACC<T, 4, 1, UseMask, UseProperties>(Q, 2, CacheProps);
+  Passed &= testACC<T, 8, 1, UseMask, UseProperties>(Q, 3, CacheProps);
+  Passed &= testACC<T, 32, 1, UseMask, UseProperties>(Q, 2, CacheProps);
```

Comment:
Please add N=16 and N=64.
If I remember it correctly, all stateless gathers (LSC and llvm.masked.gather should support non-power-of-2 lengths. Please at least one such case too.

Not sure about bti/stateful gathers, they might support non-power-of-two too.

Reply:
Sorry, I forgot it is `prefetch`, not `gather`. So, the suggestion to test non-power-of-2 may be wrong, but makes sense to check if it is supported. If Yes, then add a test case.